File: Python/Sully/Sully.py
from os import path, system


def main():
    i = 5
    source = "from os import path, system{0:c}{0:c}{0:c}def main():{0:c}    i = {1:d}{0:c}    source = {2:c}{3:s}{2:c}{0:c}    if i <= 0:{0:c}        return{0:c}    if path.exists({2:c}Sully_5.py{2:c}):{0:c}        i -= 1{0:c}    name = {2:c}Sully_{2:c} + str(i) + {2:c}.py{2:c}{0:c}    try:{0:c}        with open(name, {2:c}w{2:c}) as f:{0:c}            f.write(source.format(10, i, 34, source)){0:c}    except:{0:c}        return{0:c}    system({2:c}python3 {2:c} + name){0:c}{0:c}{0:c}if __name__ == {2:c}__main__{2:c}:{0:c}    main(){0:c}"
    if i <= 0:
        return
    if path.exists("Sully_5.py"):
        i -= 1
    name = "Sully_" + str(i) + ".py"
    try:
        with open(name, "w") as f:
            f.write(source.format(10, i, 34, source))
    except:
        return
    system("python3 " + name)


if __name__ == "__main__":
    main()

File: C/Sully/Sully.c
#include <stdio.h>
#include <unistd.h>
#include <stdlib.h>

int			main()
{
	int		i = 5;
	char	*name, *exec;
	char	*source = "#include <stdio.h>%1$c#include <unistd.h>%1$c#include <stdlib.h>%1$c%1$cint%2$c%2$c%2$cmain()%1$c{%1$c%2$cint%2$c%2$ci = %3$d;%1$c%2$cchar%2$c*name, *exec;%1$c%2$cchar%2$c*source = %4$c%5$s%4$c;%1$c%2$cFILE%2$c*f;%1$c%1$c%2$cif (i <= 0)%1$c%2$c%2$creturn (0);%1$c%2$cif (access(%4$cSully_5.c%4$c, F_OK) != -1)%1$c%2$c%2$ci--;%1$c%2$casprintf(&name, %4$cSully_%%d.c%4$c, i);%1$c%2$casprintf(&exec, %4$cgcc -Wall -Wextra -Werror %%s -o Sully_%%d && ./Sully_%%d%4$c, name, i, i);%1$c%2$cf = fopen(name, %4$cw%4$c);%1$c%2$cif (f == NULL)%1$c%2$c{%1$c%2$c%2$cfree(name);%1$c%2$c%2$cfree(exec);%1$c%2$c%2$creturn (0);%1$c%2$c}%1$c%2$cfprintf(f, source, 10, 9, i, 34, source);%1$c%2$cfclose(f);%1$c%2$csystem(exec);%1$c%2$cfree(name);%1$c%2$cfree(exec);%1$c%2$creturn (0);%1$c}%1$c";
	FILE	*f;

	if (i <= 0)
		return (0);
	if (access("Sully_5.c", F_OK) != -1)
		i--;
	asprintf(&name, "Sully_%d.c", i);
	asprintf(&exec, "gcc -Wall -Wextra -Werror %s -o Sully_%d && ./Sully_%d", name, i, i);
	f = fopen(name, "w");
	if (f == NULL)
	{
		free(name);
		free(exec);
		return (0);
	}
	fprintf(f, source, 10, 9, i, 34, source);
	fclose(f);
	system(exec);
	free(name);
	free(exec);
	return (0);
}
